Analysis/Core/CMakeLists.txt

```diff
@@ -8,10 +8,17 @@
 # granted to it by virtue of its status as an Intergovernmental Organization or
 # submit itself to any jurisdiction.
 
+find_package(FASTJET)
+
 o2_add_library(AnalysisCore
                SOURCES src/StepTHn.cxx src/CorrelationContainer.cxx
                PUBLIC_LINK_LIBRARIES O2::Framework)                  
 
+if (FASTJET_FOUND)
+  message(STATUS "Using fastjet from <${FASTJET}>")
+  target_link_libraries(O2lib-AnalysisCore PUBLIC O2::AnalysisCore)
```

Comment:
Not so sure I understand this line : was the idea to make the AnalysisCore depends on some Fastjet target here ?

Reply:
If fastjet is found, this links AnalysisCore against fastjet. Here, I assumed that we would use fastjet from AnalysisCore since this will be needed for the skimming layer.

Reply:
hum, no, that's not what that line is doing AFAICT. 

It's making the target O2lib-AnalysisCore depends (publicly) on O2::AnalysisCore. But O2::AnalysisCore and O2lib-AnalysisCore are the same targets, as O2::AnalysisCore is an alias to the other one (which should _not_ be used as is, btw, as it's kind of an internal name that might change if the o2_add_library implementation is changed at some point).

Would have imagined instead something like : 

```cmake
if (FASTJET_FOUND)
  set(FASTJETTarget somefastjet_target) 
endif()
o2_add_library(AnalysisCore 
  SOURCES src/StepTHn.cxx src/CorrelationContainer.cxx 
  PUBLIC_LINK_LIBRARIES O2::Framework ${FASTJETTarget}
)
```
Am I missing something obvious here ?

Reply:
Sorry, that line was indeed still wrong, I answered thinking of what I thought I had pushed and what is now on the branch:
```cmake
target_link_directories(O2lib-AnalysisCore PUBLIC ${FASTJET_LIBS_DIR})
```
If the target O2lib-AnalysisCore is not supposed to be used what's the point of defining them at all? The AnalysisCore get defined as an alias and is not directly usable if I am not mistaken. Maybe I am overlooking something obvious here?

Reply:
Well, cmake targets can't directly use the "namespace" syntax XX::name (only aliases can) as the actual artifact names are derived from them (e.g. libname.so for target which is a library). 

The idea is that the `o2_add_library` interface should cover most use cases so that the internal name is not actually needed, and everything should be based on the namespaced names O2::targetname instead.

In the (rare) cases that internal name _is_ actually needed, there is a `TARGETVARNAME` output parameter of the `o2_add_library` function to recover it programmatically (and thus be insulated from any change that may occur in how this internal name is derived). One such case is compile definitions, see for instance this [CMakeLists.txt](https://github.com/AliceO2Group/AliceO2/blob/dev/Detectors/ITSMFT/ITS/tracking/CMakeLists.txt)

Reply:
Ok, thanks for the clarifications and pointer. I will have a look and come up with an improved solution. I updated the FindFastJet.cmake to a target-based approach, but it still needs to be fully worked out. At least it builds and runs, so that we can work on the actual interfacing with FastJet and the jet definitions/tables in parallel.